MakieCore/src/conversion.jl

```diff
@@ -30,3 +30,103 @@ conversion_trait(::Type{<: Heatmap}) = DiscreteSurface()
 
 struct VolumeLike <: ConversionTrait end
 conversion_trait(::Type{<: Volume}) = VolumeLike()
+
+
+"""
+    convert_arguments_typed(::Type{<: AbstractPlot}, args...)::NamedTuple
+
+Converts the arguments to their correct type for the Plot type.
+Throws apropriate errors if it can't convert!
+"""
+function convert_arguments_typed end
+
+
+function convert_arguments_typed(P::Type{<: AbstractPlot}, @nospecialize(args...))
+    convert_arguments_typed(conversion_trait(P), args...)
+end
+
+function convert_arguments_typed(::CT, @nospecialize(args...)) where {CT <: ConversionTrait}
+    return convert_arguments_typed(CT, args...)
+end
+
+function convert_arguments_typed(ct::Type{<: ConversionTrait}, @nospecialize(args...))
+    # We currently just fall back to not doing anything if there isn't a convert_arguments_typed defined for certain plot types.
+    # This makes `@convert_target` an optional feature right now.
+    # It will require a bit more work to make it mandatory, and we have to think more about how to integrate it with user recipes
+    return args
+end
+
+convert_arguments_typed(::NoConversion, args...) = args
+
+"""
+    @convert_target(expr)
+
+Allows to define a conversion target for a plot type, so that convert_arguments can be checked properly, if it converts to the correct types.
+
+Usage:
+```Julia
+@convert_target struct PointBased{N} # Can be the Plot type or a conversion Trait
+    positions::AbstractVector{Point{N, Float32}}
```

Comment:
it's not clear what this does from the docstring. can this have multiple fields? are the fields just args of some function or are we making a struct here? or is the struct just a way to list the fields. that would be kind of weird